src/walk-manifest.js

```diff
@@ -49,116 +58,192 @@ var parseManifest = function(content) {
+    console.error(manifestError);
+    if (continueOnError) {
+      resources.push(manifestError);
+      return callback(null, resources);
+    }
+    return callback(manifestError);
+  }
+
+  request({url: manifest.uri, timeout: 1500}, function(error, response, body) {
+    if (error) {
+      const manifestError = new Error(error.message + '|' + manifest.uri);
+      console.error(manifestError, error);
+      if (continueOnError) {
+        resources.push(manifestError);
+        return callback(null, resources);
+      }
+      return callback(manifestError);
+    }
+    if (response.statusCode !== 200) {
+      const manifestError = new Error(response.statusCode + '|' + manifest.uri);
+      console.error(manifestError);
+      if (continueOnError) {
+        resources.push(manifestError);
+        return callback(null, resources);
+      }
+      return callback(manifestError);
+    }
+    // Only push manifest uris that get a non 200 and don't timeout
+    resources.push(manifest);
+    visitedUrls.push(manifest.uri);
+
+    manifest.content = body;
+
+    manifest.parsed = parseManifest(manifest.content);
+    manifest.parsed.segments = manifest.parsed.segments || [];
+    manifest.parsed.playlists = manifest.parsed.playlists || [];
+    manifest.parsed.mediaGroups = manifest.parsed.mediaGroups || {};
+
+    var playlists = manifest.parsed.playlists.concat(mediaGroupPlaylists(manifest.parsed.mediaGroups));
+    parseKey(basedir, decrypt, resources, manifest, parent, function(err, key) {
+      // SEGMENTS
+      manifest.parsed.segments.forEach(function(s, i) {
+        if (!s.uri) {
+          return;
+        }
+        // put segments in manifest-name/segment-name.ts
+        s.file = path.join(path.dirname(manifest.file), fsSanitize(path.basename(s.uri)));
+        if (!isAbsolute(s.uri)) {
+          s.uri = joinURI(path.dirname(manifest.uri), s.uri);
+        }
+        if (key) {
+          s.key = key;
+          s.key.iv = s.key.iv || new Uint32Array([0, 0, 0, manifest.parsed.mediaSequence, i]);
+        }
+        manifest.content = new Buffer(manifest.content.toString().replace(
+          s.uri,
+          path.relative(path.dirname(manifest.file), s.file)
+        ));
+        resources.push(s);
+      });
+
+      // SUB Playlists
+      // The reflect is used so we can still continue running even if one of the playlists is broken.
+      async.map(playlists, async.reflect(function(p, cb) {
+        if (!p.uri) {
+          return cb(null);
+        }
+        walkPlaylist(decrypt, basedir, p.uri, manifest, playlists.indexOf(p), continueOnError, cb);
+      }), function(err, reflectedResults) {
+        const results = [];
+        let callbackTopError;
+
+        reflectedResults.forEach(function(r) {
+          if (r.error && !continueOnError) {
+            console.error('Exiting early continueOnError false...|' + manifest.uri);
+            callbackTopError = r.error;
```

Comment:
So is the setup here that all of the sub playlists will always be processed regardless of the continueOnError value and then the code here will effectively promote errors that exist if continueOnError is false?

Reply:
Ya. on continueOnError is false the sublist will keep running but will just error early and not continue to recurse if they fail.

Reply:
pulling out continueOnError into the strategy style could avoid the need for this. The idea of bypassing behavior to later emulate it seems suspect.